programs/account-compression/tests/group_authority_tests.rs

```diff
@@ -0,0 +1,188 @@
+        authority: context.payer.pubkey(),
+    };
+    let instruction = Instruction {
+        program_id: account_compression::ID,
+        accounts: vec![
+            AccountMeta::new(context.payer.pubkey(), true),
+            AccountMeta::new(group_accounts.0, false),
+            AccountMeta::new_readonly(updated_keypair.pubkey(), false),
+        ],
+        data: update_group_authority_ix.data(),
+    };
+
+    let transaction = Transaction::new_signed_with_payer(
+        &[instruction],
+        Some(&context.payer.pubkey()),
+        &vec![&context.payer],
+        context.last_blockhash,
+    );
+    let update_error = context.banks_client.process_transaction(transaction).await;
+    assert!(update_error.is_err());
+
+    airdrop_lamports(&mut context, &updated_keypair.pubkey(), 1_000_000_000)
+        .await
+        .unwrap();
+    // add new program to group
+    let registered_program_pda =
+        Pubkey::find_program_address(&[compressed_pda_id.to_bytes().as_slice()], &ID).0;
+
+    let register_program_ix = account_compression::instruction::RegisterProgramToGroup {
+        program_id: compressed_pda_id,
+    };
+    let instruction = Instruction {
+        program_id: account_compression::ID,
+        accounts: vec![
+            AccountMeta::new(updated_keypair.pubkey(), true),
+            AccountMeta::new(registered_program_pda, false),
+            AccountMeta::new(group_accounts.0, false),
+            AccountMeta::new_readonly(system_program::ID, false),
+        ],
+        data: register_program_ix.data(),
+    };
+
+    let transaction = Transaction::new_signed_with_payer(
+        &[instruction],
+        Some(&updated_keypair.pubkey()),
+        &vec![&updated_keypair],
+        context.last_blockhash,
+    );
+    context
+        .banks_client
+        .process_transaction(transaction)
+        .await
+        .unwrap();
+    // add new program to group with invalid authority
+    let other_program_id = Pubkey::new_unique();
+    let registered_program_pda =
+        Pubkey::find_program_address(&[other_program_id.to_bytes().as_slice()], &ID).0;
+
+    let register_program_ix = account_compression::instruction::RegisterProgramToGroup {
+        program_id: other_program_id,
+    };
+    let instruction = Instruction {
+        program_id: account_compression::ID,
+        accounts: vec![
+            AccountMeta::new(context.payer.pubkey(), true),
+            AccountMeta::new(registered_program_pda, false),
+            AccountMeta::new(group_accounts.0, false),
+            AccountMeta::new_readonly(system_program::ID, false),
+        ],
+        data: register_program_ix.data(),
+    };
+
+    let transaction = Transaction::new_signed_with_payer(
+        &[instruction],
+        Some(&context.payer.pubkey()),
+        &vec![&context.payer],
+        context.last_blockhash,
+    );
+    context
+        .banks_client
+        .process_transaction(transaction)
+        .await
+        .unwrap_err();
```

Comment:
i suspect `.unwrap_err()` implicitly expects this to return an error?

Reply:
yes it expects to unwrap an error